meshnets/utils/data_visualization.py

```diff
@@ -7,6 +10,54 @@
 import pyvista as pv
 
 
+def plot_3d_graph_and_predictions(example,
+                                  predictions,
+                                  point_size=100,
+                                  save_path=None,
+                                  figsize=(12, 6)):
+    node_positions = example["nodes"]
+
+    fig = plt.figure(figsize=figsize)
+
+    ax1 = fig.add_subplot(121, projection="3d")
+    ax1.scatter([pos[0] for pos in node_positions],
```

Comment:
Was this done like this before?
I recall that Victor was using pyvista for these types of comparisons. However, given that these now may be graphs, it may be simpler to do them with matplotlib.

Reply:
This is supposed to work with arrays in general, couldn't just fetch the x, y, z coordinates directly from node_positions?

Reply:
> Was this done like this before?
No, before we still had the meshes stored in disk. Therefore we where plotting them using pyvisa. Now, because hugging face returns just the arrays represented as graphs we use just matplotlib.

> This is supposed to work with arrays in general, couldn't just fetch the x, y, z coordinates directly from node_positions?

If `node_positions` was an array we could use the [:,column] to extract the positions. However, directly from hugging face we are fed a `list`. For which the syntax does not work